### PR TITLE
unique keys for applied filters

### PR DESCRIPTION
**Pull request by oshi97**
I assumed that displayNames would be unique for NLP filters coming from
the API. This is not necessarily the case, and results in duplicate keys
which can result in UI bugs.

Including an underscore and index at the end makes the keys effectively unique barring
very weird edge cases.

J=none
TEST=manual

searching for Richmond on a locations page (does not exist yet, will be introduced in Alex's maps pr)
then searching for texas, does not result in an incorrect UI state

### CODE REVIEW

File: src/components/AppliedFiltersDisplay.tsx [hunk 1]
```diff
@@ -105,10 +105,11 @@ function getDedupedRemovableFilters(filters: RemovableFilter[]) {
 function renderRemovableFilter(
   displayName: string | undefined,
   handleRemove: () => void,
-  cssClasses: AppliedFiltersCssClasses
+  index: number,
+  cssClasses: AppliedFiltersCssClasses,
```
Comment (nmanu1) [inline src/components/AppliedFiltersDisplay.tsx:109]: nit: extra comma at the end

**Review by nmanu1 — DISMISSED** [review]
besides the nit, lgtm!
